### PR TITLE
Fixup distill loss - KL div scaling

### PR DESCRIPTION
**Pull request by ravwojdyla**
Some resources first:
 * pytorch implementation of `kl_div`:
    * https://github.com/pytorch/pytorch/blob/3ee75b7eacef6758db602e87287ef9574609b327/torch/nn/functional.py#L3374     
    * https://github.com/pytorch/pytorch/blob/2860f5c4f5328224ab7998bc90b9fb395b5b068b/aten/src/ATen/native/Loss.cpp#L246

We need to make sure that the 1st dimension has correct size, to get correct `kl_div`, the code before we averaging across batches only, which would sum across both sequence and vocabulary. This PR fixes it by flattening the first 2 dimensions much like we already do in the cross entropy.

### CODE REVIEW

**Review by yonromai — APPROVED** [review]
🔥
